### PR TITLE
Increase force_merge threadpool size

### LINKED ISSUE #84943: Increase force_merge thread_pool based on allocated processors
### Description

Today, [force merge is assigned one thread](https://www.elastic.co/guide/en/elasticsearch/reference/current/modules-threadpool.html) only.  

> For [force merge](https://www.elastic.co/guide/en/elasticsearch/reference/current/indices-forcemerge.html) operations. Thread pool type is fixed with a size of 1 and an unbounded queue size.

In a large cluster where there are many processors available, there could be a backlog of force merge queued up.  Perhaps we should look into changing the default of force merge thread_pool to a max of `min(1, (# of allocated processors) / 8)`

### PR DESCRIPTION
**Pull request by pxsalehi**
Changes the default size used for the force_merge threadpool to 1/8 of
the allocated processors, with a minimum value of 1.

Closes #84943

### CODE REVIEW

**Comment by pxsalehi** [comment]
@elasticsearchmachine update branch

**Comment by pxsalehi** [comment]
@elasticmachine update branch

**Comment by pxsalehi** [comment]
The idea was to also include an integration(-like) test that would pass the number of allocated processors (rather than being limited to what the Runtime returns) to the `ThreadPool`, and verify the created `force_merge` threadpool size. 

I spent some time on this, using a package private constructor to pass `allocatedProcessors` in the test. This would work for the `force_merge` threadpool, but it seems that [allocatedProcessors](https://github.com/elastic/elasticsearch/blob/898d84998b081d47ec6a9c2305f9bf4486d69be3/server/src/main/java/org/elasticsearch/common/util/concurrent/EsExecutors.java#L57) is used in other places as well to validate the `Settings` that gets passed to `ThreadPool`, e.g., in the [ExecutorBuilder](https://github.com/elastic/elasticsearch/blob/898d84998b081d47ec6a9c2305f9bf4486d69be3/server/src/main/java/org/elasticsearch/threadpool/ExecutorBuilder.java#L39). This would then make the test  fail complaining about the value passed for the `write` threadpool, for example.

So it seems to me that there needs to be more changes, to make that kind of test possible. I'm not sure, if it is worth it or not. @henningandersen WDYT?

**Comment by pxsalehi** [comment]
@elasticmachine run elasticsearch-ci/part-1

**Comment by henningandersen** [comment]
[Docs preview](https://elasticsearch_87082.docs-preview.app.elstc.co/diff)

**Comment by henningandersen** [comment]
> So it seems to me that there needs to be more changes, to make that kind of test possible. I'm not sure, if it is worth it or not. @henningandersen WDYT?

I think we should shortcut that then but at least validate that within the bounds of `[1, Runtime.availableProcessors()]`, we do get the right thread pool size. You can pick a random probe point within the range.

**Comment by pxsalehi** [comment]
> > So it seems to me that there needs to be more changes, to make that kind of test possible. I'm not sure, if it is worth it or not. @henningandersen WDYT?
> 
> I think we should shortcut that then but at least validate that within the bounds of `[1, Runtime.availableProcessors()]`, we do get the right thread pool size. You can pick a random probe point within the range.

Done. Please check https://github.com/elastic/elasticsearch/pull/87082/commits/2aa5501289a688ff369d93520f5dc7e39cb73db7.

**Review by henningandersen — APPROVED** [review]
LGTM with two small changes (provided CI passes). Thanks Pooya.

File: server/src/test/java/org/elasticsearch/threadpool/ThreadPoolTests.java [hunk 1]
```diff
@@ -51,6 +51,12 @@ public void testBoundedByBetweenMinAndMax() {
         assertThat(ThreadPool.boundedBy(value, min, max), equalTo(value));
     }
 
+    public void testOneEighthAllocatedProcessorsMinOne() {
+        assertThat(ThreadPool.oneEighthAllocatedProcessorsMinOne(4), equalTo(1));
```
[the remark this answers is not in the document]
Comment (pxsalehi) [reply server/src/test/java/org/elasticsearch/threadpool/ThreadPoolTests.java]: both done in https://github.com/elastic/elasticsearch/pull/87082/commits/34b7030126674aadb022642757360042bc597323.

File: server/src/test/java/org/elasticsearch/threadpool/ThreadPoolTests.java [hunk 2]
```diff
@@ -51,6 +51,13 @@ public void testBoundedByBetweenMinAndMax() {
         assertThat(ThreadPool.boundedBy(value, min, max), equalTo(value));
     }
 
+    public void testOneEighthAllocatedProcessorsMinOne() {
```
Comment (henningandersen) [inline server/src/test/java/org/elasticsearch/threadpool/ThreadPoolTests.java:54]: Let us also rename the test method now that the utility method has been renamed.

**Comment by pxsalehi** [comment]
Thanks, Henning!

**Pull request merged**